task.ps1

```diff
@@ -0,0 +1,133 @@
+$webHttpRule = New-AzNetworkSecurityRuleConfig -Name "web" -Description "Allow HTTP" `
+   -Access Allow -Protocol Tcp -Direction Inbound -Priority 100 -SourceAddressPrefix `
+   Internet -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 80,443
+$webNsg = New-AzNetworkSecurityGroup -ResourceGroupName $resourceGroupName -Location $location -Name `
+   $webSubnetName -SecurityRules $webHttpRule
+
+Write-Host "Creating mngSubnet network security group..."
+$mngSshRule = New-AzNetworkSecurityRuleConfig -Name "ssh" -Description "Allow SSH" `
+   -Access Allow -Protocol Tcp -Direction Inbound -Priority 100 -SourceAddressPrefix `
+   Internet -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 22
+$mngNsg = New-AzNetworkSecurityGroup -ResourceGroupName $resourceGroupName -Location $location -Name `
+   $mngSubnetName -SecurityRules $mngSshRule
+
+Write-Host "Creating a virtual network ..."
+$webSubnet = New-AzVirtualNetworkSubnetConfig -Name $webSubnetName -AddressPrefix $webSubnetIpRange -NetworkSecurityGroup $webNsg
+$mngSubnet = New-AzVirtualNetworkSubnetConfig -Name $mngSubnetName -AddressPrefix $mngSubnetIpRange -NetworkSecurityGroup $mngNsg
+$virtualNetwork = New-AzVirtualNetwork -Name $virtualNetworkName -ResourceGroupName $resourceGroupName -Location $location -AddressPrefix $vnetAddressPrefix -Subnet $webSubnet,$mngSubnet
+
+Write-Host "Creating a SSH key resource ..."
+New-AzSshKey -Name $sshKeyName -ResourceGroupName $resourceGroupName -PublicKey $sshKeyPublicKey
+
+Write-Host "Creating a web server VM ..."
+
+for (($zone = 1); ($zone -le 2); ($zone++) ) {
+   $vmName = "$webVmName-$zone"
+   New-AzVm `
+   -ResourceGroupName $resourceGroupName `
+   -Name $vmName `
+   -Location $location `
+   -image $vmImage `
+   -size $vmSize `
+   -SubnetName $webSubnetName `
+   -VirtualNetworkName $virtualNetworkName `
+   -SshKeyName $sshKeyName 
+   $Params = @{
+      ResourceGroupName  = $resourceGroupName
+      VMName             = $vmName
+      Name               = 'CustomScript'
+      Publisher          = 'Microsoft.Azure.Extensions'
+      ExtensionType      = 'CustomScript'
+      TypeHandlerVersion = '2.1'
+      Settings          = @{fileUris = @('https://raw.githubusercontent.com/mate-academy/azure_task_18_configure_load_balancing/main/install-app.sh'); commandToExecute = './install-app.sh'}
+   }
+   Set-AzVMExtension @Params
+}
+
+Write-Host "Creating a public IP ..."
+$publicIP = New-AzPublicIpAddress -Name $jumpboxVmName -ResourceGroupName $resourceGroupName -Location $location -Sku Basic -AllocationMethod Dynamic -DomainNameLabel $dnsLabel
+Write-Host "Creating a management VM ..."
+New-AzVm `
+-ResourceGroupName $resourceGroupName `
+-Name $jumpboxVmName `
+-Location $location `
+-image $vmImage `
+-size $vmSize `
+-SubnetName $mngSubnetName `
+-VirtualNetworkName $virtualNetworkName `
+-SshKeyName $sshKeyName `
+-PublicIpAddressName $jumpboxVmName
```

Comment:
It is not required to create this resources.

Reply:
fix